src/shell/shell_batch.cpp

```diff
@@ -19,275 +19,211 @@
 #include "shell.h"
 
 #include <climits>
-#include <stdlib.h>
-#include <string.h>
+#include <cstdlib>
+#include <cstring>
 
 #include "logging.h"
 #include "string_utils.h"
 
 // Permitted ASCII control characters in batch files
-constexpr uint8_t BACKSPACE = 8;
+constexpr uint8_t BACKSPACE       = 8;
 constexpr uint8_t CARRIAGE_RETURN = '\r';
-constexpr uint8_t ESC = 27;
-constexpr uint8_t LINE_FEED = '\n';
-constexpr uint8_t TAB = '\t';
-constexpr uint8_t UNIT_SEPARATOR = 31;
-
-BatchFile::BatchFile(DOS_Shell *host,
-                     char const *const resolved_name,
-                     char const *const entered_name,
-                     char const *const cmd_line)
-        : file_handle(0),
-          location(0),
-          echo(host->echo),
+constexpr uint8_t ESC             = 27;
+constexpr uint8_t LINE_FEED       = '\n';
+constexpr uint8_t TAB             = '\t';
+constexpr uint8_t UNIT_SEPARATOR  = 31;
+
+[[nodiscard]] static bool FoundLabel(std::string_view line, std::string_view label);
+
+BatchFile::BatchFile(DOS_Shell* host, const char* const resolved_name,
+                     const char* const entered_name, const char* const cmd_line)
+        : echo(host->echo),
           shell(host),
           prev(host->bf),
-          cmd(new CommandLine(entered_name, cmd_line)),
-          filename("")
+          cmd(new CommandLine(entered_name, cmd_line))
 {
-	char totalname[DOS_PATHLENGTH+4];
+	char totalname[DOS_PATHLENGTH + 4];
 
 	// Get fullname including drive specification
-	if (!DOS_Canonicalize(resolved_name, totalname))
+	if (!DOS_Canonicalize(resolved_name, totalname)) {
 		E_Exit("SHELL: Can't determine path to batch file %s", resolved_name);
-	
+	}
+
 	filename = totalname;
 	// Test if file is openable
-	if (!DOS_OpenFile(totalname,(DOS_NOT_INHERIT|OPEN_READ),&file_handle)) {
-		//TODO Come up with something better
-		E_Exit("SHELL:Can't open BatchFile %s",totalname);
+	if (!DOS_OpenFile(totalname, (DOS_NOT_INHERIT | OPEN_READ), &file_handle)) {
+		// TODO Come up with something better
+		E_Exit("SHELL:Can't open BatchFile %s", totalname);
```

Comment:
Should be a warning, not an error, and the message should be:

```
SHELL: Can't open batch file '%s'
```

You could do a cleanup pass of the log messages too if you like.